### ISSUE TITLE
New Package Request: ArchiveBox

### ISSUE THREAD

**Issue by pirate** [NixOS/nixpkgs#122793]
**Project description**

ArchiveBox is a self-hosted internet archiving solution that lets you save sites offline.

**Metadata**

- homepage URL: https://archivebox.io
- source URL: https://github.com/ArchiveBox/ArchiveBox
- license: MIT
- platforms: [Ubuntu/Debian](https://github.com/ArchiveBox/debian-archivebox), [Arch](https://aur.archlinux.org/packages/archivebox/), [FreeBSD](https://github.com/ArchiveBox/ArchiveBox/blob/dev/bin/setup.sh), [macOS](https://github.com/ArchiveBox/homebrew-archivebox), [Windows](https://archivebox.io/#quickstart) (WSL/WSL2/Docker only)

**Dependencies**

- `python` >= 3.7
- `django` >=3.1,<3.2
- `node` >= 12
- `chromium` / `chrome` >= 70
- `wget` >= 1.2
- `curl` >= 7.3
- `youtube-dl`
- `git`

For more info, see: https://github.com/ArchiveBox/ArchiveBox#dependencies and https://github.com/ArchiveBox/ArchiveBox/wiki/Install#1-install-dependencies

**Context**

I'm the maintainer of ArchiveBox, but I'm a NixOS n00b so I don't feel qualified to package it for Nix, but I'd be happy to help anyone that wants to.

**Comment by vojta001** [comment]
@pirate I am volunteering to do it. I have made a WIP version and one thing complicates the packaging process a bit – your django version requirement. Will things break if we supply django `3.2` (instead of django `<3.2,>=3.1.3`)?

**Comment by vojta001** [comment]
I made a WIP PR with django 3.2, but archivebox fails with
```
ValueError: Primary key 'django.db.models.UUIDField' referred by core.apps.CoreConfig.default_auto_field must subclass AutoField.
```

**Comment by pirate** [comment]
Ah yeah 3.2 is not supported yet, there are a number of incompatibilities besides that one ^. IS it possible to vendor/pin an older version of django? I'm about to do this for the debian package as well.

**Comment by siraben** [comment]
Closed by #133476

**Comment by siraben** [comment]
I only learned about the django failure after it was merged into Nixpkgs 😅. Anyway I fixed it in #133483 and it's working for me now.

**Comment by mweinelt** [comment]
How are y'all using the package, given that we don't have a module? I'm not keen on reinventing one.
